### PR TITLE
feat: Add Sampling Decision to Trace Envelope Header

### LINKED ISSUE #3201: Add Sampling Decision to Trace Envelope Header
### Description

See project description: https://github.com/getsentry/team-sdks/issues/8

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Adds `sampled` flag to all trace envelope headers

## :bulb: Motivation and Context

closes #3201

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Sentry/include/SentryTraceContext.h [hunk 1]
```diff
@@ -52,6 +52,11 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nullable, nonatomic, strong) NSString *sampleRate;
 
+/**
+ * Value indicating whether the trace was sampled.
+ */
+@property (nullable, nonatomic, strong) NSString *sampled;
```
Comment (armcknight) [inline Sources/Sentry/include/SentryTraceContext.h:58]: These properties should probably be `readonly` and/or `copy` if they're only settable via the init method

**Review by armcknight — APPROVED** [review]
Changes look fine. Though I wonder if an `NSNumber` encapsulating a boolean wouldn't be more appropriate than a string. 

If we want to leave it as a string, we should consider using defined constants for the values.

File: Sources/Sentry/Public/SentryDefines.h [hunk 2]
```diff
@@ -22,6 +22,9 @@
     -(instancetype)init NS_UNAVAILABLE;                                                            \
     +(instancetype) new NS_UNAVAILABLE;
 
+#define SENTRY_TRUE_STRING @"true"
+#define SENTRY_FALSE_STRING @"false"
```
Comment (brustolin) [inline Sources/Sentry/Public/SentryDefines.h:26]: @armcknight Do you think a `#define` is fine or a `const` would be better?

I will create another PR later to replace every instance of "true" and "false"

Comment (armcknight) [reply Sources/Sentry/Public/SentryDefines.h:26]: I prefer `static NSString *const` for these over macros, but it's not a huge deal.

**Review by armcknight — APPROVED** [review]
Updates look good, thanks!

**Pull request merged**